plugins/vault-backend/src/service/vaultApi.ts

```diff
@@ -88,11 +93,58 @@ export interface VaultApi {
 export class VaultClient implements VaultApi {
   private vaultConfig: VaultConfig;
   private readonly limit = plimit(5);
+  private token: string = '';
 
   constructor(options: { config: Config }) {
     this.vaultConfig = getVaultConfig(options.config);
   }
 
+  static async fromConfig(config: Config): Promise<VaultApi> {
+    const vaultClient = new VaultClient({ config });
+
+    if (typeof vaultClient.vaultConfig.token === 'string') {
+      vaultClient.token = vaultClient.vaultConfig.token;
+    } else {
+      vaultClient.token = await vaultClient.kubernetesLogin(
```

Comment:
can this token expire? If yes, maybe we should perform the login ondemand? This would solve the issue of my previous comment

Reply:
I think with the new approach, this should be fixed now. You can use together the kubernetes login with the refresh token. So it should be solved